// frog/_lib/utils/requestBodyToComposerActionBaseContext.js
import { getRequestUrl } from './getRequestUrl.js';
import { verifyComposerAction } from './verifyComposerAction.js';
export async function requestBodyToComposerActionBaseContext(c, { hub, verify = true, verifyOrigin = true, }) {
    const { trustedData, untrustedData } = (await c.req.json().catch(() => { })) || {};
    const url = getRequestUrl(c.req);
    const trustedComposerActionData = await (async () => {
        if (verify === false)
            return null;
        if (!trustedData)
            return null;
        if (!hub)
            return null;
        try {
            const { composerActionData } = await verifyComposerAction({
                hub,
                frameUrl: untrustedData.url,
                trustedData,
                url: url.href,
                verifyOrigin,
            });
            return composerActionData;
        }
        catch (err) {
            if (verify === 'silent')
                return null;
            throw err;
        }
    })();
    return {
        env: c.env,
        actionData: trustedComposerActionData || untrustedData,
        req: c.req,
        var: c.var,
        verified: Boolean(trustedComposerActionData),
    };
}
